finetuner/tuner/dataset/__init__.py

```diff
@@ -1,8 +1,183 @@
+import abc
 import itertools
+from typing import Any, Callable, Dict, Generic, List, Optional, Tuple, TypeVar, Union
 
 import numpy as np
 
 from ... import __default_tag_key__
+from ...helper import DocumentSequence
+
+
+AnyLabel = TypeVar('AnyLabel')
+
+
+class BaseDataset(abc.ABC, Generic[AnyLabel]):
```

Comment:
Do these `Generic` really work on `Python`? or is it only like a type hint?

Reply:
They are for type hinting yes